crates/flox-rust-sdk/src/models/environment/generations.rs

```diff
@@ -0,0 +1,328 @@
+    /// and registered as a new generation using [Self::add_generation].
+    pub fn realize_generation(
+        &self,
+        generation: usize,
+    ) -> Result<PathEnvironment<Original>, GenerationsError> {
+        let realized_gens = self.realize()?;
+
+        let tempdir = tempfile::tempdir_in(&self.tempdir_base)
+            .unwrap()
+            .into_path();
+        let environment = PathEnvironment::new(
+            realized_gens.path().join(generation.to_string()),
+            self.pointer.clone(),
+            tempdir,
+            Original,
+        )
+        .unwrap();
+
+        Ok(environment)
+    }
+
+    /// Realize the current generation as set in the metadata file
+    /// as a [PathEnvironment].
+    ///
+    /// The generation can then be safely modified
+    /// and registered as a new generation using [Self::add_generation].
+    pub fn realize_current_generation(
+        &self,
+    ) -> Result<PathEnvironment<Original>, GenerationsError> {
+        let metadata = self.metadata()?;
+        let current_gen = metadata
+            .current_gen
+            .ok_or(GenerationsError::NoGenerations)?;
+        self.realize_generation(*current_gen)
+    }
+
+    /// Import an existing environment into a generation
+    ///
+    /// Assumes the invariant that the [PathEnvironment] instance is valid.
+    ///
+    /// This will copy the manifest and lockfile from the source environment
+    /// into a generation folder.
+    /// Any other assets such as hook scripts are ignored.
+    ///
+    /// If the generation already exists, it will be overwritten.
+    ///
+    /// If `set_current` is true, the generation will also be set as the current generation.
+    fn register_generation(
+        &mut self,
+        environment: PathEnvironment<Original>,
+        generation: usize,
+        generation_metadata: GenerationMetadata,
+        set_current: bool,
+    ) -> Result<(), GenerationsError> {
+        let realized = self.realize()?;
+
+        let description = generation_metadata.description.clone();
+
+        let mut metadata = self.metadata()?;
+        let _existing = metadata
+            .generations
+            .insert(generation.into(), generation_metadata);
+
+        if set_current {
+            metadata.current_gen = Some(generation.into());
+        }
+
+        write_metadata_file(metadata, realized.path())?;
+
+        let generation_path = realized.path().join(generation.to_string());
+        let env_path = generation_path.join(ENV_DIR_NAME);
+        fs::create_dir_all(&env_path).unwrap();
+
+        // todo: we need to figure out what comes with a path env.
+        //       i.e. which files other than the manifest and lockfile do we need to copy?
+        //       what about assets such hook scripts?
+
+        // copy manifest and lockfile (if it exists)
+        // silently overrides existing files (but does not remove them if they are not in the source)
```

Comment:
Question: what does this comment mean by "overrides existing files"? Didn't we just create the `env_path` that these are being copied into?

Reply:
This method does not check whether the environment already existed,
you can give any generation number and it will just copy into the existing generations folder.

with add_generation this wont hapen as that guarantees sequential generations